src/parser/mod.rs

```diff
@@ -6249,6 +6249,11 @@ impl<'a> Parser<'a> {
                 loc
             );
         }
+        // Mysql requires table specification on index drop
+        let table = match self.parse_keyword(Keyword::ON) {
+            true => Some(self.parse_object_name(false)?),
+            false => None,
+        };
```

Comment:
```suggestion
        let table = if self.parse_keyword(Keyword::ON) {
            Some(self.parse_object_name(false)?)
        } else {
            None
        };
```

Reply:
@iffyio thanks for the review! I've applied your suggestion